### DISCUSSION TITLE
Combo box "active state" identifer

### DISCUSSION THREAD

**Discussion by mdurchholz** [uswds/uswds#5948] [Ideas]
It would be ideal if the combo box parent wrapper had some kind of class or attribute that toggles when the element is active. Does anyone have a solution to this right now? I am unsure how to track when the list items receive focus. I have looked into mutation observers, but I was curious if there was a more simple solution. I do not want to sole rely on focus of the input field since that input element loses focus once you enter the options menu.

EDIT:
An "on active/unactive" event watcher would be incredible as well.

**Comment by mejiaj** [comment]
@mdurchholz can you share a little more information on what you're trying to do? Screenshots would help, but any additional information would be appreciated!

**Comment by mdurchholz** [comment]
@mejiaj - I am trying to add a class to combo box parent wrapper whenever a user is interacting with it:

<img width="524" alt="Screenshot 2024-06-28 at 9 47 30 AM" src="https://github.com/uswds/uswds/assets/3687595/139a19b5-01c0-4053-b1fe-b7735861429a">

<br>
Something like the below would suffice:

```
<div class="usa-combo-box usa-combo-box__menu-open"  data-enhanced="true">
    ...
</div>
```

**Comment by mdurchholz** [comment]
I think it is also worth saying that a dev shouldn't have to wrap the default select element in a ".usa-combo-box" parent in order for the scripts to trigger. I feel like that HTML should be dynamically added in if it does not exist. Imagine if you are adding this into an existing site. ALL of the select elements throughout the site would need to be updated.

**Comment by mejiaj** [comment]
@mdurchholz thanks for that information. I've created an issue for this enhancement - #5969.

Feel free to add any additional information in the issue. Additionally feel free to close this discussion if you feel it's resolved.

**Comment by mejiaj** [comment]
Closing because we've created an issue for it.
